app/assets/javascripts/questions.js

```diff
@@ -30,16 +30,25 @@ function initQuestionFilter() {
   function showAll() {
     $('.my-question').removeClass('hidden');
     $('.other-question').removeClass('hidden');
+    $('.no-questions').addClass('hidden');
   }
 
   function showMine() {
-    $('.my-question').removeClass('hidden');
-    $('.other-question').addClass('hidden');
+     if ($('.my-question').length == 0) {
+
+      $('.no-questions').removeClass('hidden');
+      $('.other-question').addClass('hidden');
+    }
+    else {
+       $('.no-questions').addClass('hidden');
+       $('.other-question').addClass('hidden');
```

Comment:
This line is common to both if and else blocks, so it can be moved outside of the if block.

A very succinct way to accomplish all of this (test it first please!) would be:

```
$('.no-questions').toggleClass('hidden', $('.my-question').length > 0);
$('.other-question').addClass('hidden');
```
also, where did the `$('.my-question').removeClass('hidden');` go? Don't we need to show the `.my-question` rows still?